clientd/src/server/mod.rs

```diff
@@ -0,0 +1,64 @@
+use crate::{Event, EventKey, ManagerMessage, RpcRequest};
+use jsonrpsee::server::ServerBuilder;
+use jsonrpsee::RpcModule;
+use std::collections::HashMap;
+use std::net::SocketAddr;
+use std::sync::Arc;
+use tokio::sync::{broadcast, mpsc, oneshot};
+
+pub struct Context {
+    manager_tx: Arc<mpsc::Sender<ManagerMessage>>,
+}
+fn register_methods(module: &mut RpcModule<Context>) -> anyhow::Result<()> {
+    module.register_async_method("health_check", |_, context| async move {
+        let (cbtx, cbrx) = oneshot::channel();
+        let rpc_req = ManagerMessage::CallHandler(RpcRequest::HealthCheck(), cbtx);
+        context.manager_tx.send(rpc_req).await.unwrap();
+        Ok(cbrx.await.unwrap())
+    })?;
+
+    module.register_async_method("info", |_, context| async move {
+        let (cbtx, cbrx) = oneshot::channel();
+        let rpc_req = ManagerMessage::CallHandler(RpcRequest::Info(), cbtx);
+        context.manager_tx.send(rpc_req).await.unwrap();
+        Ok(cbrx.await.unwrap())
+    })?;
+    Ok(())
+}
+fn register_subscriptions(
+    _module: &mut RpcModule<Context>,
+    _se_map: HashMap<EventKey, broadcast::Sender<Event>>,
```

Comment:
I'm pretty sure you'll want an `Arc` around that eventually or for it to live inside a context object with an `Arc` around it. You'll see it when you get there though, probably ok for now.

Reply:
kind of like this :
```
txOne = se_map.get(EventKey::One).unwrap().clone()
module.add_subscription(idk, move |_,_| async  {
    // we dont have a reference to se_map neither did we move it 
})
```
(the syntax is probably not correct but you get the gist)

Reply:
Yeah, that's a lot of deep cloning. Works, but not elegant imo.